### ISSUE TITLE
Desktop: Entire text is not selected while tabbing between textfields

### ISSUE THREAD

**Issue by KaushikGupta007** [flutter/flutter#150339]
### Steps to reproduce

1. Create two Text Fields on Home page
2. Run app on windows
3. Write text in first and second field
4. Click on first field
5. Press Tab key on keyboard to move to second field
6. Text is not selected in second field

### Expected results

Text in Field should be selected when moving to next field

### Actual results

Text in Text Field in not selected

### Code sample

<details open><summary>Code sample</summary>

```dart
Future<void> main() async {
  runApp(MaterialApp(
    initialRoute: "/",
    routes: {
      "/": (context) => const Home(),
    },
  ));
}

class Home extends StatefulWidget {
  const Home({super.key});

  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          children: [
            TextFormField(
              decoration: const InputDecoration(hintText: "First Field", labelText: "First Field")
            ),
            const SizedBox(height: 8,),
            TextFormField(
              decoration: const InputDecoration(hintText: "Second Field", labelText: "Second Field")
            )
          ],
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

**Expected Result in Flutter Web**

https://github.com/flutter/flutter/assets/129219828/f8107e40-09f0-4984-864e-b2bdd7e5c092

**Actual Result in Flutter Windows**

https://github.com/flutter/flutter/assets/129219828/2f71dd06-5517-4108-ab3a-52725596298d

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Version 10.0.22621.3593], locale en-IN)
    • Flutter version 3.22.2 on channel stable at C:\Users\Admin\fvm\versions\3.22.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (12 days ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at C:\Users\Admin\AppData\Local\Android\sdk
    • Platform android-34, build-tools 33.0.1
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.1)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34616.47
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2022.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[√] VS Code (version 1.89.1)
    • VS Code at C:\Users\Admin\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.90.0

[√] Connected device (5 available)
    • CPH2467 (mobile)                   • 1a715d74      • android-arm64  • Android 14 (API 34)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.22621.3593]
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 126.0.6478.61
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 125.0.2535.85

[√] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by justinmc** [comment]
We should probably confirm what the native behavior is on Windows, Linux, and MacOS. If anyone knows for certain and has tried it on any of these platforms (not in a web browser), please confirm whether text gets selected or not when tabbing. I think Flutter's current behavior is to preserve the selection.

@darshankawar Did you try this on a native Mac app or just on a Flutter Mac app?

| Native Windows | Native Mac | Native Linux |
| --- | --- | --- |
| Selects all | ❓ | ❓ |

**Comment by darshankawar** [comment]
> @darshankawar Did you try this on a native Mac app or just on a Flutter Mac app?

@justinmc Hi, I tried now using Reminders app on Mac and it seem to select the entire text upon tabbing:


https://github.com/flutter/flutter/assets/67046386/0943d0c3-81fe-4c1c-8c0f-d7af53219c59

**Comment by gnprice** [comment]
For Linux, I tried just now in the GNOME Settings application, and its behavior was to select the whole text. So that's probably the normal behavior for GNOME-based applications, at least.

**Comment by KaushikGupta007** [comment]
Is this fixed in the latest 3.24 release?

**Comment by KaushikGupta007** [comment]
Hi team,

I wanted to follow up on this issue to see if there has been any progress or if additional information is needed from my side to help with the resolution. I understand that you're likely busy with many other requests, but I'd appreciate any updates or guidance on next steps.

Thank you for your time and support!
